### PR TITLE
Use lambda coefficient in back-translation step

### PR DESCRIPTION
**Pull request by ethanjperez**
`lambda_coeff` is currently unused only in the back-translation step (`bt_step`) of training. In the other training steps (i.e. `mt_step`), `lambda_coeff` is used to weight the loss.

### CODE REVIEW

**Comment by glample** [comment]
Thank you!
